### ISSUE TITLE
E2531 Node.18x deprecation for function creation does not match official AWS docs

### ISSUE THREAD

**Issue by cullancarey** [aws-cloudformation/cfn-lint#4249]
### CloudFormation Lint Version

cfn-lint 1.40.0

### What operating system are you using?

Ubuntu

### Describe the bug

# E2531 Node.js 18.x deprecation dates in cfn-lint do not match official AWS Lambda docs

## Description
When running `cfn-lint` version `1.40.0`, the following error is raised:

```
E2531 Runtime ‘nodejs18.x’ was deprecated on ‘2025-07-31’.
Creation was disabled on ‘2025-10-01’ and update on ‘2025-11-01’.
Please consider updating to ‘nodejs22.x’
```

However, the official AWS Lambda runtime lifecycle documentation lists different deprecation and block dates for Node.js 18.x:

| Name       | Identifier | OS             | Deprecation date | Block function create | Block function update |
|------------|------------|----------------|------------------|-----------------------|-----------------------|
| Node.js 18 | nodejs18.x | Amazon Linux 2 | Sep 1, 2025      | Feb 3, 2026           | Mar 9, 2026           |

## Expected Behavior
`cfn-lint` should align with the AWS published deprecation and block dates:
- Deprecation date: **Sep 1, 2025**
- Block function creation: **Feb 3, 2026**
- Block function update: **Mar 9, 2026**

## Actual Behavior
`cfn-lint` reports earlier deprecation and block dates:
- Deprecation: **Jul 31, 2025**
- Creation disabled: **Oct 1, 2025**
- Update disabled: **Nov 1, 2025**

## References
- AWS Lambda Runtime Support Policy: [AWS Docs](https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html#runtimes-deprecated)  
- `cfn-lint` runtime lifecycle JSON: [LmbdRuntimeLifecycle.json#L80](https://github.com/aws-cloudformation/cfn-lint/blob/v1.40.0/src/cfnlint/data/AdditionalSpecs/LmbdRuntimeLifecycle.json#L80)

## Environment
- cfn-lint version: `1.40.0`

## Expected Behavior
`cfn-lint` should align with the AWS published deprecation and block dates:
- Deprecation date: **Sep 1, 2025**
- Block function creation: **Feb 3, 2026**
- Block function update: **Mar 9, 2026**

### Reproduction template

```
  LambdaFunction:
    Type: "AWS::Serverless::Function"
    Properties:
      CodeUri: functions/function
      Description: ""
      FunctionName: name
      Handler: 'index.handler'
      Role: role
      Runtime: nodejs18.x
      Timeout: 50

**Comment by cullancarey** [comment]
Opened a [PR](https://github.com/aws-cloudformation/cfn-lint/pull/4250) to fix the dates.
